bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -626,10 +632,8 @@ private void notifyPrimaryAddressChange(@Nullable String oldPrimaryAddress, @Nul
         }
     }
 
-    private boolean getConfigParameter(Map<String, Object> parameters, String parameter, boolean defaultValue) {
-        if (parameters == null) {
-            return defaultValue;
-        }
+    private boolean getConfigParameter(Map<String, @Nullable Object> parameters, String parameter,
```

Comment:
Without that annotation, the compiler assumes that a `get()` call returns a non-null value and thus you will always get wrong warnings in the following code (because you obviously have to check for a null value). With this annotation, the compiler knows that a result can be null and thus provides the correct hints in the following code.

Reply:
That's caused by the missing EEAs.

```Java
T get(K key)
```

is part of an non-annotated class because we lack the EEA for `Map`.

Because of the missing EEA the compiler uses the annotation of `T` for the return value.
But this is not correct.
The EEA should add this annotation:

```Java
@Nullable T get(K key)
```

Regardless if T is nullable or not, get will return null for an unknown key.

With your annotation you state that the map is allowed to contain nullable values.
It makes e.g. a different for `put`, is null allowed or not.

We should not change the semantic of the map just because we don't use EEA.

Reply:
@maggu2810 Why did I change any semantic? The specification says that the parameter is a `java.util.Map`, it does not say that values are non-null (or did I overlook this information?).
I think it is therefore
1. correct.
2. anyhow not relevant for `put` as this is no valid/meaningful operation on a map that is passed in by DS.
3. solving the issue that the compiler adds many warnings when doing necessary null checks, which can easily seduce people to remove that code, although it is needed and correct.

I do not see any cons, only pros.

Reply:
Anyhow, in order to speed up the merge of this critical fix, I have removed the annotation fixes with https://github.com/eclipse/smarthome/pull/6091/commits/cd3db27facc8b7f44aebf4090f1ae9258a8284f0, so we can discuss this separately.